src/containers/Main.js

```diff
@@ -18,51 +18,49 @@ import {StyleProvider} from "../contexts/StyleContext";
 import "./Main.css";
 import Profile from "./profile/Profile";
 
-export default class Main extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isDark: false
-    };
-  }
+const Main = () => {
+  const [isDark, setIsDark] = useState(false);
 
-  componentDidMount() {
+  useEffect(() => {
     if (localStorage.getItem("isDark") === null) {
       const darkPref = window.matchMedia("(prefers-color-scheme: dark)");
       localStorage.setItem("isDark", darkPref.matches);
     }
-    this.setState({isDark: JSON.parse(localStorage.getItem("isDark"))});
-  }
-  changeTheme = () => {
-    this.setState({isDark: !this.state.isDark}, () => {
-      localStorage.setItem("isDark", this.state.isDark);
-    });
+    setIsDark(JSON.parse(localStorage.getItem("isDark")));
+  }, []);
+
+  const changeTheme = () => {
+    setIsDark(!isDark);
   };
+  useEffect(() => {
+    // Update local storage as soon as isDark changes
+    localStorage.setItem("isDark", isDark);
```

Comment:
Why do we need a separate useEffect for this? Moving this localStorage call to the `changeTheme` function should handle this in `StyleProvider` ?

Reply:
You can abstract all of this away in a custom hook (probably should create a utils or hooks folder) called [`useLocalStorage()`](https://usehooks.com/useLocalStorage). This would also solve the problem of using the context API to get style state as we can switch to using only local storage calls.

I know this might add a bit more work than your original proposed changes but this seems like the most optimal and simplest solution going forward.